pmlc/ast/builder.cc

```diff
@@ -607,6 +608,53 @@ struct ProgramBuilder {
         .result();
   }
 
+  Value makeGatherOp(ExprNodeIntrinsic *node, ArrayRef<Value> operands) {
+    auto operands_size = operands.size();
+    TensorShape shape = evaluator.getShape(node);
+    RankedTensorType resultType = builder.getRankedTensorType(shape);
+    auto op = builder.create<tile::GatherOp>(
+        loc, resultType, ValueRange{operands[0], operands[1]},
+        ArrayRef<NamedAttribute>{});
+
+    switch (operands_size) {
+    case 5: {
```

Comment:
What are these magic numbers?

Reply:
Basically, I don't think we should try to do different logic based on different numbers of operands at this stage. Can we force the C API to always supply the full number of operands and default those to sensible values above?

Reply:
Changed to fixed number (5) of operands for gather.